models/src/main/kotlin/motif/models/Dependencies.kt

```diff
@@ -43,8 +44,24 @@ class Dependencies(val clazz: IrClass, val scope: Scope) {
     companion object {
 
         fun fromScope(scope: Scope): Dependencies? {
-            val dependenciesClass = scope.clazz.annotatedInnerClass(motif.Dependencies::class) ?: return null
+            val creatable = findCreatableSuperinterface(scope.clazz) ?: return null
+            val dependenciesType = creatable.typeArguments.singleOrNull() ?: return null
+            val dependenciesClass = dependenciesType.resolveClass() ?: return null
             return Dependencies(dependenciesClass, scope)
         }
+
+        private fun findCreatableSuperinterface(clazz: IrClass): IrClass? {
+            if (clazz.qualifiedName.takeWhile { it != '<' } == Creatable::class.java.name) {
+                return clazz
+            }
+
+            clazz.supertypes
+                    .mapNotNull { it.resolveClass() }
+                    .forEach { superinterface ->
+                        findCreatableSuperinterface(superinterface)?.let { return it }
+                    }
+
+            return null
```

Comment:
Is there a test case for no `Creatable` parent? Doesn't seem so from this PR alone.

Reply:
There are but I'll add one that tests this case explicitly.